### PR TITLE
feat: send contract verification packets

### PR DESCRIPTION
**Pull request by colinnielsen**
## Checklist

- [ ] Documentation updated
- [ ] Tested code changes

[ticket]()

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@0xmetropolis/metal",
-  "version": "0.8.3",
+  "version": "0.9.0",
   "description": "Smart contract visualization tool.",
```
Comment (kevinweaver) [inline package.json:4]:
```suggestion
  "description": "Smart contract deployment tool.",
```

File: src/commands/preview.ts [hunk 2]
```diff
@@ -123,6 +129,14 @@ export const configureForgeScriptInputs = ({ rpcUrl }: { rpcUrl: string }): stri
   if (!forgeArguments.includes('--broadcast')) forgeArguments.push('--broadcast');
   if (!forgeArguments.includes('--slow')) forgeArguments.push('--slow');
 
+  if (forgeArguments.includes('--verify') || forgeArguments.includes('--verifier-url'))
+    throw Error(
+      'Must verify contracts against the metal backend, please do not include --verify or --verifier-url flags',
+    );
+  forgeArguments.push(
+    `--verify --verifier-url ${METAL_SERVICE_URL}/contracts/verification/${previewId} --etherscan-api-key null`,
```
Comment (kevinweaver) [inline src/commands/preview.ts:137]: is `--etherscan-api-key null` necessary

File: src/utils/foundry.ts [hunk 3]
```diff
@@ -246,7 +247,20 @@ const watchForgeOutput = (
   const magicEmojis = ['🧙', '🪄', '🧚', '✨'];
 
   stdout.on('data', (chunk: any) => {
-    // if the chunk contains the succes label, then it's related to a transaction
+    if (state.disableLogging) {
+      if (!chunk.toString().includes('contracts were verified!')) return;
+      // resume logging
+      else state.disableLogging = false;
+    }
```
Comment (kevinweaver) [inline src/utils/foundry.ts:254]: ❓ What happens when contracts fail to verify and `contracts were verified!` never appears? Does the UX make sense if logging is disabled and is never re-enabled?

**Review by its-everdred — APPROVED** [review]

**Comment by kevinweaver** [comment]
Approved to merge after integration tests pass

File: package.json (hunk 1 above)
Comment (colinnielsen) [reply package.json:4]: heh

File: src/commands/preview.ts (hunk 2 above)
Comment (colinnielsen) [reply src/commands/preview.ts:137]: kind of - when type of verifier is etherscan, yes

File: src/utils/foundry.ts (hunk 3 above)
Comment (colinnielsen) [reply src/utils/foundry.ts:254]: the verification request always returns a success message

**Pull request merged**